app/controllers/routes_controller.rb

```diff
@@ -11,9 +11,19 @@ def show
 
   def update
     route_details = @request_data[:route]
-    @route = Route.find_or_initialize_by(incoming_path: route_details.delete(:incoming_path))
-    status_code = @route.new_record? ? 201 : 200
-    @route.update_attributes(route_details) || status_code = 422
+    incoming_path = route_details.delete(:incoming_path)
+    tries = 3
+    begin
+      @route = Route.find_or_initialize_by(incoming_path: incoming_path)
+      status_code = @route.new_record? ? 201 : 200
+      @route.update_attributes(route_details) || status_code = 422
+    rescue Moped::Errors::OperationFailure => e
+      if e.details["code"] == 11000 && (tries -= 1) > 0
```

Comment:
This is nitpicking, but I can only guess this code relates to the unique index violation from mongo because I'm reading this PR and have context on it right now.  As whoever comes to this next probably won't have that context can we give the code a name via a constant or something?